tests/Resources/ti.ui.navigationwindow.test.js

```diff
@@ -9,45 +9,184 @@
 'use strict';
 const should = require('./utilities/assertions');
 
-describe.windowsMissing('Titanium.UI.NavigationWindow', function () {
+describe('Titanium.UI.NavigationWindow', function () {
 	this.timeout(10000);
 
 	let nav;
-	afterEach(function () {
-		if (nav) {
-			nav.close();
+	afterEach(done => {
+		if (nav && !nav.closed) {
+			nav.close().then(() => done()).catch(_e => done());
+		} else {
+			nav = null;
+			done();
 		}
-		nav = null;
 	});
 
-	it.iosBroken('Ti.UI.NavigationWindow', () => { // should this be defined?
+	it.iosBroken('namespace exists', () => { // should this be defined?
 		should(Ti.UI.NavigationWindow).not.be.undefined();
 	});
 
-	it('.apiName', function () {
-		var view = Ti.UI.createNavigationWindow();
-		should(view).have.readOnlyProperty('apiName').which.is.a.String();
-		should(view.apiName).be.eql('Ti.UI.NavigationWindow');
+	describe('properties', () => {
+		it('.apiName', () => {
+			const view = Ti.UI.createNavigationWindow();
+			should(view).have.readOnlyProperty('apiName').which.is.a.String();
+			should(view.apiName).be.eql('Ti.UI.NavigationWindow');
+		});
 	});
 
-	it('#open()', function () {
-		var view = Ti.UI.createNavigationWindow();
-		should(view.open).be.a.Function();
-	});
+	describe('methods', () => {
+		describe('#close()', () => {
+			it('is a Function', () => {
+				const nav = Ti.UI.createNavigationWindow();
 
-	it('#openWindow()', function () {
-		var view = Ti.UI.createNavigationWindow();
-		should(view.openWindow).be.a.Function();
-	});
+				should(nav).have.a.property('close').which.is.a.Function();
+			});
 
-	it('#close()', function () {
-		var view = Ti.UI.createNavigationWindow();
-		should(view.close).be.a.Function();
-	});
+			it('returns a Promise', finish => {
+				nav = Ti.UI.createNavigationWindow({ window: Ti.UI.createWindow() });
+
+				const openPromise = nav.open();
+				openPromise.then(() => {
+					const result = nav.close();
+					result.should.be.a.Promise();
+					result.then(() => finish()).catch(e => finish(e));
+				}).catch(e => finish(e));
+			});
+
+			it('called on unopened Window rejects Promise', finish => {
+				nav = Ti.UI.createNavigationWindow({ window: Ti.UI.createWindow() });
+
+				const result = nav.close();
+				result.should.be.a.Promise();
+				result.then(() => finish(new Error('Expected #close() to be rejected on unopened Window'))).catch(_e => finish());
+			});
+
+			it('called twice on Window rejects second Promise', finish => {
+				nav = Ti.UI.createNavigationWindow({ window: Ti.UI.createWindow() });
+
+				nav.open().then(() => {
```

Comment:
<!--
  0 failure: 
  1 warning:  tests/Resources/t...
  
  
  DangerID: danger-id-Danger;
  File: tests/Resources/ti.ui.navigationwindow.test.js;
  Line: 67;
-->

- :warning: tests/Resources/ti.ui.navigationwindow.test.js line 67 – Each then() should return a value or throw (promise/always-return)